### ISSUE TITLE
Need ability to specify command timeout

### ISSUE THREAD

**Issue by kmc059000** [AMVSoftware/Cake.SqlServer#12]
I am having issues with timeouts because the default value for SqlCommand.CommandTimeout is 30 seconds. There is currently no way to set it through the Cake.SqlServer API.

**Comment by trailmax** [comment]
That's an interesting problem. 
Can you please give a bit more context why you need such a long timeout?

**Comment by kmc059000** [comment]
I'm attempting to backup and restore a decently sized database in my script.

**Comment by trailmax** [comment]
Do you mind sharing how you do restoring? See issue #7 - I'm a bit stuck deciding about best course of action. Any opinion will be good to hear there.

And yes, I'll add timeout optional parameter sometime this week.

**Comment by kmc059000** [comment]
In a cake task, I do the following.
```
var script = string.Format(@"
        USE MASTER
        ALTER DATABASE [{0}]
        SET SINGLE_USER
        WITH ROLLBACK IMMEDIATE;

        GO

        RESTORE DATABASE  [{0}]
        FROM DISK = '{1}'
        WITH REPLACE {2}

        ALTER DATABASE [{0}]
        SET MULTI_USER
        WITH ROLLBACK IMMEDIATE;

        GO", destinationDatabaseName, backupFileName, restoreConditions);

CreateDatabaseIfNotExists(destinationConnString, destinationDatabaseName);
ExecuteSqlCommand(destinationConnString, script);
```

`restoreConditions` is an optional string for additional restore parameters. It is not very elegant but gets the job done in my case. I think it the addin should definitely do it differently.

```
var restoreConditions = @",
  MOVE 'db' TO 'C:\Program Files\Microsoft SQL Server\MSSQL10_50.MSSQLSERVER\MSSQL\DATA\db-restore.mdf',
  MOVE 'db_log' TO 'C:\Program Files\Microsoft SQL Server\MSSQL10_50.MSSQLSERVER\MSSQL\DATA\db-restore.ldf'"
```

**Comment by kmc059000** [comment]
I'm also working on a PR at the moment for you to look at. It's up to you how to proceed, but I have something that works, and does not cause backwards compatibility issues. I'll explain more once I submit it.

**Comment by trailmax** [comment]
reopening so I don't forget about DB-restore comment.

**Comment by trailmax** [comment]
backup restore is now implemented in v1.5.0.
